### PR TITLE
feat: adding side navigation component

### PR DESCRIPTION
**Pull request by bclark-p44**
#DES-426
#DES-450

## 📝 Description

Adding menu and navigation components. Needed to update the provider to support a theme in order to support our movement application. This will come in handy as we add new themes and color modes!

## Screenshots

> Please provide screenshots for any visual changes

## Merge checklist

- [ ] Added/updated tests
- [ ] Added changeset

### CODE REVIEW

File: packages/navigation/tests/Navigation.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { render, screen } from '@testing-library/react';
```
Comment (danielrobertson) [inline packages/navigation/tests/Navigation.stories.tsx:1]: @branclar Should this file be named `Navigation.test.tsx` not `Navigation.stories.tsx`? A proposal for fix in https://github.com/project44/manifest/pull/216

Comment (bclark-p44) [reply packages/navigation/tests/Navigation.stories.tsx:1]: yep! Will update

File: packages/menu/src/MenuGroup/MenuGroup.context.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { createContext } from '@project44-manifest/react-utils';
+
+export interface MenuGroupContext {
+  isGrouped: boolean;
+}
+
+export const [MenuGroupProvider, useMenuGroup] = createContext<MenuGroupContext>({
+  name: 'MenuGroupContext',
+});
```
Comment (danielrobertson) [inline packages/menu/src/MenuGroup/MenuGroup.context.ts:9]: nice https://github.com/danielrobertson/manifest/blob/main/packages/utils/src/context.ts#L18

**Review by danielrobertson — APPROVED** [review]

**Review by andrea-p44 — CHANGES_REQUESTED** [review]
Looks like the px between menu items is 8px and should be 4px. And sub-items in the menu should be regular weight text.

**Pull request merged**